src/App.tsx

```diff
@@ -1,29 +1,70 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+
+type Intial<T extends number[]> = {
+  total: number;
+  perPageOptions: T;
+  perPageInitial: T[number];
+};
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+
+const intialData: Intial<[3, 5, 10, 20]> = {
+  total: 42,
+  perPageOptions: [3, 5, 10, 20],
+  perPageInitial: 5,
+};
+
+const items = getNumbers(1, intialData.total).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(intialData.perPageInitial);
+  const [activePage, setActivePage] = useState(1);
+
+  const firstItemOnPage = perPage * activePage - perPage + 1;
+  const lastItemOnPage =
+    perPage * activePage > intialData.total
+      ? intialData.total
+      : perPage * activePage;
+
+  const getCurrentItems = (): string =>
+    `${firstItemOnPage} - ${lastItemOnPage}`;
+
+  function handleSelectEvent(event: React.ChangeEvent<HTMLSelectElement>) {
+    setActivePage(1);
+    setPerPage(
+      Number(event.target.value) as (typeof intialData.perPageOptions)[number],
+    );
+  }
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${activePage} (items ${getCurrentItems()} of ${items.length})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            onChange={handleSelectEvent}
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            value={perPage}
+          >
+            {[...intialData.perPageOptions].map(opt => {
```

Comment:
Using `Math.random()` for keys in React is not recommended as it will generate new keys on every render, leading to performance issues. It's better to use a unique and consistent identifier, such as the option value itself.